File: core/src/main/java/org/strykeforce/thirdcoast/telemetry/util/SineSignalGenerator.java
package org.strykeforce.thirdcoast.telemetry.util;

class SineSignalGenerator extends SignalGenerator {

  public SineSignalGenerator(double frequency, double phase, double amplitude, double offset,
      double invert) {
    super(frequency, phase, amplitude, offset, invert);
  }

  @Override
  double getValue(double time) {
    double t = frequency * time + phase;
    double val = Math.sin(2.0 * Math.PI * t);
    return invert * amplitude * val + offset;
  }

  @Override
  public String toString() {
    return "Sine at " + super.toString();
  }
}

File: src/main/kotlin/org/strykeforce/telemetry/util/SineSignalGenerator.kt
package org.strykeforce.telemetry.util

internal class SineSignalGenerator(
    frequency: Double, phase: Double, amplitude: Double, offset: Double, invert: Double
) : SignalGenerator(frequency, phase, amplitude, offset, invert) {

    override fun getValue(time: Double): Double {
        val t = frequency * time + phase
        val `val` = Math.sin(2.0 * Math.PI * t)
        return invert * amplitude * `val` + offset
    }

    override fun toString(): String {
        return "Sine at " + super.toString()
    }
}
